### PR TITLE
Various build fixes

### PR DESCRIPTION
**Pull request by jakirkham**
Makes various improvements to the build like using the `toolchain`, not using Homebrew to install `gmp`, and avoiding `LD_LIBRARY_PATH` as it is forbidden by SIP on macOS 10.12.

### CODE REVIEW

**Comment by jakirkham** [comment]
cc @dougalsutherland

File: recipe/build.sh [hunk 1]
```diff
@@ -1,18 +1,17 @@
...
 then
-    export LD_LIBRARY_PATH=$PREFIX/lib:$LD_LIBRARY_PATH
-    if [ $TRAVIS ];
-    then
-        brew uninstall gmp
-    fi
+    export LDFLAGS="${LDFLAGS} -Wl,-rpath,${PREFIX}/lib"
+elif [ "$(uname)" == "Linux" ];
+then
+    export LDFLAGS="${LDFLAGS} -Wl,-rpath=${PREFIX}/lib"
```
Comment (dougalsutherland) [inline recipe/build.sh]: Why repeat the LDFLAGS code here? Actually, `build.sh` will only ever run on either Darwin or Linux, so....

**Comment by dougalsutherland** [comment]
Cool, LGTM other than that one comment.

File: recipe/build.sh (hunk 1 above)
Comment (jakirkham) [reply recipe/build.sh]: So they are actually slightly different. After `rpath`, Linux uses `=` and macOS uses `,`. Admittedly a bit subtle and seemingly needless difference. Unfortunately it is a difference we need to distinguish between.

Comment (dougalsutherland) [reply recipe/build.sh]: Oh, I missed that. Oh well.

**Comment by jjhelmus** [comment]
LGTM

**Comment by jakirkham** [comment]
Thanks!

**Pull request merged**